pygac/utils.py

```diff
@@ -20,100 +20,43 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 import gzip
+import io
 import logging
-import numpy as np
-import sys
-from contextlib import contextmanager
-
-LOG = logging.getLogger(__name__)
-
 
-def is_file_object(filename):
-    """Check if the input is a file object.
+from contextlib import contextmanager, nullcontext
 
-    Args:
-        filename - object to check
-
-    Note:
-        This method only check if the object implements the
-        interface of a file object to allow duck types like
-        gzip.GzipFile instances.
-    """
-    has_close = hasattr(filename, 'close')
-    has_read = hasattr(filename, 'read')
-    if hasattr(filename, 'seekable'):
-        is_seekable = filename.seekable()
-    else:
-        is_seekable = False
-    return has_close and has_read and is_seekable
+import numpy as np
 
+LOG = logging.getLogger(__name__)
 
-@contextmanager
-def _file_opener(file):
-    """Open a file depending on the input.
 
-    Args:
-        file - path to file or file object
-    """
-    # open file if necessary
-    if is_file_object(file):
-        open_file = file
-        close = False
-    else:
-        open_file = open(file, mode='rb')
-        close = True
-    # check if it is a gzip file
+def gzip_inspected(open_file):
+    """Try to gzip decompress the file object if applicable."""
     try:
-        file_object = gzip.open(open_file)
-        file_object.read(1)
+        file_object = gzip.GzipFile(mode='rb', fileobj=open_file)
+        file_object.read1()
```

Comment:
Actually, this method exists, but it is not what I thought it is, instead of reading one byte, it reads with a single call to the unbuffered stream... Fix is coming.